src/HealthChecks.Uris/UriHealthCheck.cs

```diff
@@ -55,6 +59,25 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                             return new HealthCheckResult(context.Registration.FailureStatus, description: $"Discover endpoint #{idx} is not responding with code in {expectedStatusCodes.Min}...{expectedStatusCodes.Max} range, the current status is {response.StatusCode}.");
                         }
 
+                        if (expectedContent != null)

```

Comment:
You can do all in the same condition? 
if ( expctedContent != null || expected ContentFunc!=null)
va expected = expectedContent ?? expectedContentFunc()

Reply:
I think on first pass you might have misunderstood the `Func` usage. The signature is `Func<HttpContent, Task<ContentCheckResult>>`. It gets given the content to check, rather than just being a method that returns a string for the expected content.

So your comment here is irrelevant. Will leave for you to resolve.